novatel_gps_driver/src/nodelets/novatel_gps_nodelet.cpp

```diff
@@ -356,6 +369,51 @@ namespace novatel_gps_driver
     }
 
     /**
+        * SystemAlert to shutdown the sensor node
+        */
+      void alertCallback(const cav_msgs::SystemAlertConstPtr &msg)
+       {
+         if( msg->type==cav_msgs::SystemAlert::FATAL || msg->type==cav_msgs::SystemAlert::SHUTDOWN)
+           {
+            ros::shutdown();
```

Comment:
Please log something here so that the reason for the node shutdown is recorded.

Reply:
You can just check for SHUTDOWN here. Checking for FATAL might cause this node to shutdown too early in the event of a system failure when a recovery component still needs it.

Reply:
ok